File: backend/app/schemas/user_management.py
from pydantic import BaseModel, ConfigDict, EmailStr, Field


class UserManagementResponse(BaseModel):
    id: int
    name: str
    email: EmailStr
    role_id: int
    role: str
    is_active: bool

    model_config = ConfigDict(
        from_attributes=True,
    )


class UserListResponse(BaseModel):
    users: list[UserManagementResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class UserCreateRequest(BaseModel):
    name: str = Field(
        min_length=2,
        max_length=100,
    )

    email: EmailStr

    password: str = Field(
        min_length=6,
        max_length=128,
    )

    role: str = Field(
        default="USER",
        min_length=1,
        max_length=50,
    )

    is_active: bool = True


class UserUpdateRequest(BaseModel):
    name: str | None = Field(
        default=None,
        min_length=2,
        max_length=100,
    )

    email: EmailStr | None = None

    role: str | None = Field(
        default=None,
        min_length=1,
        max_length=50,
    )


class UserStatusRequest(BaseModel):
    is_active: bool

File: frontend/src/types/userManagement.ts
export interface ManagedUser {
  id: number;
  name: string;
  email: string;
  role_id: number;
  role: string;
  is_active: boolean;
}

export interface UserListResponse {
  users: ManagedUser[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface UserCreateRequest {
  name: string;
  email: string;
  password: string;
  role: string;
  is_active: boolean;
}

export interface UserUpdateRequest {
  name?: string;
  email?: string;
  role?: string;
}

export interface UserStatusRequest {
  is_active: boolean;
}

export interface UserQueryParams {
  page?: number;
  page_size?: number;
  search?: string;
  role?: string;
  is_active?: boolean;
}